Tasks/Maven/maventask.ts

```diff
@@ -212,11 +213,17 @@ function createMavenSQRunner(sqHostUrl, sqHostUsername, sqHostPassword, sqDbUrl?
 function applyPmdGoals(mvnRun: trm.ToolRunner):void {
     if (tl.getInput('pmdAnalysisEnabled', true) != 'true') {
         console.log('PMD analysis is not enabled');
+        return;
     }
     console.log('PMD analysis is enabled');
     pmd.applyPmdArgs(mvnb);
 }
 
+// Take a result object and create one line of the build summary from it.
+function buildSummaryLineFromResult(analysisResult:ar.AnalysisResult):string {
+    return `${analysisResult.toolName} found ${analysisResult.totalViolations} violations in ${analysisResult.filesWithViolations} files.`;
+}
+
```

Comment:
Resolved. Done.

In the event of a build or test failure, Maven default is to fail immediately and cease executing all further goals. As the code analysis goals are deliberately added at the very end, this will result in code analysis results being unavailable in the result of failures in build or test - it is considered that the user will want to fix the build or test failure(s) first.